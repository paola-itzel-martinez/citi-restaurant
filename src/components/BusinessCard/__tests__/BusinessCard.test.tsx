import React from "react";
import {
  YELP_FIXTURES,
  customRender,
  fireEvent,
  screen,
} from "../../../../tests";
import { BusinessCard } from "..";

const [BUSINESS] = YELP_FIXTURES.BUSINESSES;
const VIEW_BUTTON_TEXT = "View";
const WINDOW_OPEN_FN = jest.fn();

window.open = WINDOW_OPEN_FN;

const GeneralComponent = () => <BusinessCard {...BUSINESS} />;

describe("<BusinessCard />", () => {
  it('Should show all data with specific styles rules', () => {
    const { container } = customRender(<GeneralComponent />);

    expect(screen.getByText(BUSINESS.name)).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it('Should open a new tab when view button is clicked', () => {
    customRender(<GeneralComponent />);

    fireEvent.click(screen.getByText(VIEW_BUTTON_TEXT));

    expect(WINDOW_OPEN_FN).toBeCalledTimes(1);
  });
});
