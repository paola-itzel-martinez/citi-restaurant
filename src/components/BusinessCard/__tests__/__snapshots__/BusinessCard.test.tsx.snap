// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BusinessCard /> Should show all data with specific styles rules 1`] = `
.c13 {
  background: linear-gradient(115deg,#ff6f04,#ffc391);
  border: unset;
  border-radius: 8px;
  color: #ffffff;
  font-family: 'Roboto',sans-serif;
  font-size: 16px;
  font-weight: 400;
  text-transform: uppercase;
  box-shadow: unset;
  padding: 8px 32px;
  white-space: nowrap;
}

.c13.selected,
.c13:hover {
  background: #ff6f04;
  border: unset;
  cursor: pointer;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  padding: 28px;
  gap: 16px;
  background: #ffffff;
  box-shadow: 0px 0.25px 1px rgba(0,0,0,0.039),0px 0.85px 3px rgba(0,0,0,0.19);
  border-radius: 8px;
}

.c0:hover {
  cursor: pointer;
}

.c7 {
  border-color: #e0e6eb;
  border-style: solid;
  border-width: 0px 0px thin;
  -webkit-align-self: unset;
  -ms-flex-item-align: unset;
  align-self: unset;
  height: unset;
  width: 100%;
  margin-block: unset;
  margin-inline: unset;
}

.c2 {
  margin-inline: undefined;
  margin-top: undefined;
  margin-right: undefined;
  margin-bottom: undefined;
  margin-left: undefined;
}

.c10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 2px;
}

.c11 {
  display: inline-block;
  color: #ffc104;
  overflow: hidden;
  width: 16px;
  text-overflow: clip;
}

.c6 {
  color: #415058;
  font-family: 'Roboto',sans-serif;
  font-weight: 400;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  font-size: 16px;
  text-overflow: ellipsis;
  overflow: hidden;
  white-space: nowrap;
  padding: 0;
  margin: 0;
  min-width: 20px;
  margin-inline: undefined;
  margin-top: undefined;
  margin-right: undefined;
  margin-bottom: undefined;
  margin-left: undefined;
}

.c12 {
  color: #415058;
  font-family: 'Roboto',sans-serif;
  font-weight: 400;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  font-size: 14px;
  padding: 0;
  margin: 0;
  min-width: 20px;
  margin-inline: undefined;
  margin-top: undefined;
  margin-right: undefined;
  margin-bottom: undefined;
  margin-left: undefined;
}

.c1 {
  position: relative;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 200px;
}

.c14 {
  position: absolute;
  left: 50%;
  bottom: -16px;
  -webkit-transform: translateX(-50%);
  -ms-transform: translateX(-50%);
  transform: translateX(-50%);
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 16px;
  width: 100%;
}

.c4 .c5 {
  text-align: center;
  width: 100%;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  gap: 16px;
}

.c8.final-divider {
  display: none;
}

.c3 {
  width: 125px;
  height: 115px;
}

@media (min-width:0px) and (max-width:599px) {
  .c13 {
    white-space: unset;
  }
}

@media (min-width:0px) and (max-width:599px) {
  .c0 {
    box-shadow: unset;
    padding: 0;
    width: 100%;
  }
}

@media (min-width:0px) and (max-width:599px) {
  .c6 {
    font-size: 20px;
  }
}

@media (min-width:0px) and (max-width:599px) {
  .c12 {
    font-size: 14px;
  }
}

@media (min-width:0px) and (max-width:599px) {
  .c1 {
    position: unset;
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
    -webkit-flex-wrap: wrap;
    -ms-flex-wrap: wrap;
    flex-wrap: wrap;
    -webkit-align-items: strech;
    -webkit-box-align: strech;
    -ms-flex-align: strech;
    align-items: strech;
    width: 100%;
  }
}

@media (min-width:0px) and (max-width:599px) {
  .c14 {
    position: unset;
    left: unset;
    bottom: unset;
    -webkit-transform: unset;
    -ms-transform: unset;
    transform: unset;
  }
}

@media (min-width:0px) and (max-width:599px) {
  .c4 .c5 {
    text-align: start;
  }
}

@media (min-width:0px) and (max-width:599px) {
  .c4 {
    -webkit-align-items: flex-start;
    -webkit-box-align: flex-start;
    -ms-flex-align: flex-start;
    align-items: flex-start;
    -webkit-box-pack: justify;
    -webkit-justify-content: space-between;
    -ms-flex-pack: justify;
    justify-content: space-between;
    -webkit-flex: 1;
    -ms-flex: 1;
    flex: 1;
    gap: 12px;
    width: 40%;
  }
}

@media (min-width:0px) and (max-width:599px) {
  .c9 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    gap: 6px;
  }
}

@media (min-width:0px) and (max-width:599px) {
  .c8.final-divider {
    display: block;
  }
}

@media (min-width:0px) and (max-width:599px) {
  .c8.data-divider {
    display: none;
  }
}

@media (min-width:0px) and (max-width:599px) {
  .c3 {
    width: calc(45% - 16px);
    min-width: 115px;
    max-width: 115px;
  }
}

<div>
  <div
    class="c0 c1"
    id="i50BEr4Pq6-D8tmsegmVyA"
    name="The Table"
  >
    <img
      alt="The Table"
      class="c2 c3"
      draggable="false"
      src="https://s3-media4.fl.yelpcdn.com/bphoto/Y70NLnzrDZ-vkW9xiTAM1Q/o.jpg"
    />
    <div
      class="c4"
    >
      <p
        class="c5 c6"
        color="#415058"
      >
        The Table
      </p>
      <hr
        class="c7 c8 data-divider"
      />
      <div
        class="c9"
      >
        <div
          class="c10"
        >
          <i
            class="c11 fas fa-star"
            data-testid="filled-star"
          />
          <i
            class="c11 fas fa-star"
            data-testid="filled-star"
          />
          <i
            class="c11 fas fa-star"
            data-testid="filled-star"
          />
          <i
            class="c11 fas fa-star"
            data-testid="filled-star"
          />
          <i
            class="c11 far fa-star"
            data-testid="empty-star"
          />
        </div>
        <p
          class="c5 c12"
          color="#415058"
        >
          $$
        </p>
      </div>
      <button
        class="c13 c14"
      >
        View
      </button>
    </div>
  </div>
  <hr
    class="c7 c8 final-divider"
  />
</div>
`;
